components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt

```diff
@@ -240,6 +263,20 @@ class PromptFeature private constructor(
                         prompt.dismiss()
                     }
                     activePrompt?.clear()
+                    loginPicker.dismissCurrentLoginSelect()
+                }
+        }
+
+        // Dismiss prompts when a new tab is selected.
+        sessionPromptScope = store.flowScoped { flow ->
+            flow.ifChanged { browserState -> browserState.selectedTabId }
```

Comment:
This would mean that we only be notified for selected sessions and not custom tabs, is that right? I think your previous version that looking for the session ID would have worked too, just the `ifChanged` needed to be simplified from: 
`ifChanged { it.engine.engineSession }` to just `ifChanged()`.

Reply:
Yeah I think selected sessions ignoring custom tabs should be what we want here! And this should also resolve the issue here with other prompts showing again after switching tabs https://github.com/mozilla-mobile/fenix/issues/12943#issuecomment-669069988

Reply:
That's fair. It would isolate our Sync users that sign-in (which uses a Custom Tab) who may have sync'd with logins, signed out, and then attempted to sign it again using the saved logins, but maybe that's too fine of an edge case.